src/RegistrationController.js

```diff
@@ -137,6 +137,9 @@ function (
             ele.addClass('password-complexity-meet');
           } else {
             ele.removeClass('password-complexity-meet');
+            if(alert) {
+              ele.addClass('password-complexity-alert');
```

Comment:
You should be to simplify the logic by limiting to only have one `class` value.
Also the `blur` event always happen after `input` event. That could have impact on your flow too.